src/python/pants/backend/python/tasks2/pex_build_util.py

```diff
@@ -41,10 +41,15 @@ def has_python_requirements(tgt):
 def dump_sources(builder, tgt, log):
   buildroot = get_buildroot()
   log.debug('  Dumping sources: {}'.format(tgt))
-  for relpath in tgt.sources_relative_to_source_root():
+  for relpath in tgt.sources_relative_to_buildroot():
     try:
-      src = os.path.join(buildroot, tgt.target_base, relpath)
-      if isinstance(tgt, Resources):
+      if type(tgt) == Files:
+        src = os.path.join(buildroot, relpath)
+      else:
+        relpath = os.path.relpath(relpath, tgt.target_base)
+        src = os.path.join(buildroot, tgt.target_base, relpath)
+
+      if has_resources(tgt):
```

Comment:
These target-type checks being inside the loop is a bit odd... worthwhile to move them out?